javaslang/src/main/java/javaslang/collection/HashSet.java

```diff
@@ -26,7 +26,7 @@
  * @author Ruslan Sennov, Patryk Najda, Daniel Dietrich
  * @since 2.0.0
  */
-public final class HashSet<T> implements Kind1<HashSet<?>, T>, Set<T>, Serializable {
+public final class HashSet<T> extends AbstractSet<T> implements Kind1<HashSet<?>, T>, Serializable {
```

Comment:
Inheriting from a package-private (see comment above) subclass causes trouble. Java 8 has a bug regarding method-references and inherited public impls of invisible super-classes...

A solution could be to create a helper class `Sets` (similar to `Maps` and `Arrays`) and add a methods

```
class Sets {
    static boolean equals(Set<?> set1, Object o) {
        if (set1 == o) {
            return true;
        } else if (o instanceof Set) {
            final Set<?> set2 = (Set<?>) o;
            ...
        } else {
            return false;
        }
    }
}
```